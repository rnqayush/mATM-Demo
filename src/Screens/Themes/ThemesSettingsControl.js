import React from 'react';
import useThemes from '@optimus/themes/useThemes';
import Labels from 'translations/settingsScreen/labels';
import { ButtonActionRow } from 'screens/settingsScreen/ButtonActionRow';
import { INITIAL_THEME_STATE, THEME_SETTINGS, THEME_KEY } from './ThemesConstants';

const ThemesSettingsControl = props => {
  const { selectedThemeName } = useThemes();
  const state = {
    ...INITIAL_THEME_STATE,
    [THEME_KEY]: Labels[selectedThemeName],
  };

  return <ButtonActionRow {...props} item={THEME_SETTINGS} state={state} />;
};

export default ThemesSettingsControl;
