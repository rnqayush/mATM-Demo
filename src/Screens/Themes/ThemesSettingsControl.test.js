import React from 'react';
import { render } from 'test/test-utils';
import ThemesSettingsControl from './ThemesSettingsControl';

jest.mock('@optimus/themes/useThemes', () => () => ({ selectedThemeName: 'LIGHT' }));
jest.mock('screens/settingsScreen/ButtonActionRow', () => ({ ButtonActionRow: () => <></> }));

describe('ThemesSettingsControl', () => {
  it('should render ThemesSettingsControl component', () => {
    const component = render(<ThemesSettingsControl />);
    expect(component).toMatchSnapshot();
  });
});
