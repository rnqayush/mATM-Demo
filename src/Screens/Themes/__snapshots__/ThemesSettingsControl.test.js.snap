// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemesSettingsControl should render ThemesSettingsControl component 1`] = `null`;
